### PR TITLE
Edge cases

### PR DESCRIPTION
**Pull request by kprazuch**
## Link to Ticket
https://trello.com/c/0YIrsbaC/170-backend-edge-case-checks

## Brief Description of Changes

- Added try, except blocks to all API calls
- Added conditions to check if errors are returned by API calls
- Updated existing tests and added new tests

Note: Currently logging all errors to a file but I wasn't sure if that was the best way of doing it. So it's kind of a temp solution at the moment.

## Checklist
- [x] I do not commit the .env file.
- [x] I added my changes to the appropriate frontend or backend folders.
- [x] All aspects of the ticket have been entirely addressed and completed.
- [x] Thorough automated tests have been added.
- [x] Functionality has been validated through manual tests.
- [ ] I have received approval from a peer before merging.

### CODE REVIEW

File: backend/app.py [hunk 1]
```diff
@@ -21,6 +22,11 @@
 app = Flask(__name__)
 CORS(app)
 
+logging.basicConfig(filename="warnings",
+                    filemode='a',
+                    format='%(asctime)s,%(msecs)d %(name)s %(levelname)s %(message)s',
+                    datefmt='%H:%M:%S')
```
Comment (tzada8) [inline backend/app.py:28]: Logging all errors to a file could be a good temp solution as long as we can still see this file for the deployed site.

**Review by tzada8 — APPROVED** [review]
Looks great! Now I think the frontend should (hopefully) never crash. Worst case scenario no data will be returned, but at least the app will still run
